src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -89,7 +92,50 @@ public sealed class CompletionItem : IComparable<CompletionItem>
         /// <summary>
         /// Additional information attached to a completion item by it creator.
         /// </summary>
-        public ImmutableDictionary<string, string> Properties { get; }
+        public ImmutableDictionary<string, string> Properties
+        {
+            get
+            {
+                if (_propertiesAsImmutableDictionary is null)
+                    _propertiesAsImmutableDictionary = _properties.ToImmutableDictionary();
+
+                return _propertiesAsImmutableDictionary;
+            }
+        }
+
+        internal IEnumerable<KeyValuePair<string, string>> GetProperties()
```

Comment:
>GetProperties [](http://example.com/codeflow?start=59&length=13)

I didn't want to expose the ImmutableArray directly (as exposing that publicly is why I can't get rid of the ImmutableDictionary). Most callers should suffice with the (Try)GetProperty calls, if not, they can feel the pain of a boxed enumerator.